File: solutions/007/007.c
#include <stdio.h>

int main()
{
  int Primes[10001];
  Primes[0] = 2;
  Primes[1] = 3;
  int Position;
  int Check = 5;

  for ( int PrimesIndex = 2; PrimesIndex < 10001; PrimesIndex++ )
  {
    for ( Position = 1; Primes[Position]*Primes[Position] <= Check; Position++ )
      if ( !(Check % Primes[Position]) )
      {
        Position = 0;
        Check += 2;
      }
    Primes[PrimesIndex] = Check;
    Check += 2;
  }

  printf( "%d\n", Primes[10000] );
}

File: solutions/007/007.py
primes=[i for i in range(1,10002)]
primes[0] = 2
primes[1] = 3
check = 5
position = 1

for prime_index in range(2, 10001):
    while primes[position-1]*primes[position] <= check:
        if not check % primes[position]:
            position = 1
            check += 2
        else:
            position += 1
    position = 1
    primes[prime_index] = check
    check += 2

print(primes[10000])
